### PR TITLE
Exploring Matplotlib for Data Visualization in Python  

### PR DESCRIPTION
**Pull request by lalith1403**
I have written a full fledged article covering numpy, matplotlib and use cases of the respective libraries. I have given snippets of code explaining the features of the library. I have checked for grammar on grammarly.com There is one additional image that i have included a link to, along witht the hero.png file. 

description: exploring matplotlib features for data visualization in python. We cover the various use cases and generate 2-D, 3-D plots and various charts for data analysis. 
author_name: Lalithnarayan C

### CODE REVIEW

**Comment by hectorkambow** [comment]
@lalith1403  it seems like this PR has the other two articles on the same branch -if you could edit to make sure every article has its own separate branch 👍

**Comment by lalith1403** [comment]
> @lalith1403 it seems like this PR has the other two articles on the same branch -if you could edit to make sure every article has its own separate branch 👍

I have removed the files... it includes only one article. Let me know if everything is alright now

**Comment by hectorkambow** [comment]
Good afternoon Mani 👋 @manibatra  - wanted to ask if you could give this article a tech review before we publish.
Thank you 🚀

File: articles/exploring-matplotlib-for-visualizations-in-python/index.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+
+#### Coding Time
+
+Let's begin by creating a numpy array
+
+```python
+import numpy as np
+a=np.array([1,2,3,4,5])
+```
+The input to the np.array should be a list. To specify the dtype at runtime, we manually add the argument dtype in the np.array function.
```
Comment (manibatra) [inline articles/exploring-matplotlib-for-visualizations-in-python/index.md:33]: `To specify the data type at runtime, we manually add the argument dtype in the np.array function.` makes more sense to me.

File: articles/exploring-matplotlib-for-visualizations-in-python/index.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+The above functions take in the array size, in the form of a tuple as an input.
+
+Creating equally spaced lists is a usual problem during plotting. Therefore, we use linspace or arange to achieve the same.
+
+```python
+import numpy as np
+spacedArray = np.linspace(0,5,10)
+# equally spaced array with gap of 0.5
+```
+Using linspace is better than arange. arange misbehaves when presented with floating-point numbers.
```
Comment (manibatra) [inline articles/exploring-matplotlib-for-visualizations-in-python/index.md:56]: Should highlight `linspace` and `arange` as they are functions and not just normal words in the language. Should do this for other functions/commands throughout the article.

**Comment by hectorkambow** [comment]
@lalith1403 - this article is having a merge conflict - we believe due to the FORK not being up to date with the EngEd repo. 

If you could try and rebase the PR to make sure everything is caught up:
Rule of thumb:
Should always be branching from their master
Should always keep their fork up to date with section-io/engineering-education

These commands should help in git:

# only need to do this once
git remote add section http://github.com/section-io/engineering-education 

# then every time
git fetch section
git pull --rebase section master
git push origin

**Comment by hectorkambow** [comment]
Article is live 
https://www.section.io/engineering-education/matplotlib-visualization-python/

**Pull request closed without merging**